darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -81,17 +83,28 @@ def get_encoded_fig(mpl_fig: Any):
         encoded_fig = base64.b64encode(tmpfile.getvalue()).decode("utf-8")
         return encoded_fig
 
-    def generate_img(self):
+    def generate_fig(self):
         """
-        Generate the image using the figure data.
+        Generate a figure using the figure data.
         """
-        # generate the matplotlib figure using the figure's
+        # generate the figure using the figure's
         # function and function arguments
-        mpl_fig = self.fig_func(**self.fig_args)
-        # encode the matplotlib figure
-        encoded = self.get_encoded_fig(mpl_fig=mpl_fig)
-        # create the img string
-        self.img_str = f"<img src=data:image/png;base64,{encoded} alt={self.fig_title} width={self.fig_width}>"
+        fig = self.fig_func(**self.fig_args)
+        supported_mpl_fig_types = [matplotlib.figure.Figure, seaborn.axisgrid.JointGrid]
+        if type(fig) in supported_mpl_fig_types:
```

Comment:
I'm not ecstatic about this bit because the _real_ requirement is that the figure has a `.savefig()` method, so this might be overly strict. But I think we do want to catch cases where incorrect figure types are generated instead of failing at `.savefig()` downstream..

Reply:
Why not use the real requirement then? This will let things that quack like a duck through.

```diff
--- a/darshan-util/pydarshan/darshan/cli/summary.py
+++ b/darshan-util/pydarshan/darshan/cli/summary.py
@@ -13,8 +13,6 @@ from typing import Any, Union, Callable
 
 import pandas as pd
 from mako.template import Template
-import matplotlib
-import seaborn
 
 import darshan
 import darshan.cli
@@ -90,8 +88,7 @@ class ReportFigure:
         # generate the figure using the figure's
         # function and function arguments
         fig = self.fig_func(**self.fig_args)
-        supported_mpl_fig_types = [matplotlib.figure.Figure, seaborn.axisgrid.JointGrid]
-        if type(fig) in supported_mpl_fig_types:
+        if hasattr(fig, "savefig"):
             # encode the matplotlib figure
             encoded = self.get_encoded_fig(mpl_fig=fig)
             # create the img string
@@ -101,8 +98,7 @@ class ReportFigure:
             self.fig_html = fig.to_html(index=False, border=0)
         else:
             err_msg = (
-                f"Figure of type {type(fig)} not supported. \n"
-                f"Supported figure types: {supported_mpl_fig_types}"
+                f"Figure of type {type(fig)} not supported."
             )
             raise NotImplementedError(err_msg)
```

Reply:
Thanks for the suggestion. It was sort of an afterthought that I forgot to circle back to, but I'll implement these changes.